cloudbuild.yaml

```diff
@@ -200,13 +212,26 @@ steps:
     args:
       - '-c'
       - |
-        tar zcf cache.tgz .gradle/caches .gradle/wrapper
+        if [ -f $(< hashsum)-cache.tgz ]; then
+          echo "Cache already exists, skipping build cache"
+        else
+          tar zcf cache.tgz .gradle/caches .gradle/wrapper
+        fi
 
   - name: 'gcr.io/cloud-builders/gsutil'
     id: &save_cache 'Save gradle cache to GCS'
+    entrypoint: 'bash'
     waitFor:
       - *compress_cache
-    args: [ 'cp', 'cache.tgz', 'gs://${_CACHE_BUCKET}/cache.tgz' ]
+    args:
+      - '-c'
+      - |
+        if [ -f $(< hashsum)-cache.tgz ]; then
+          echo "Cache already exists, skipping save cache"
+        else
+          # TODO: Do not persist indefinitely. Add a retention policy to GCS bucket.
```

Comment:
Could you create a bug for this and reference as TODO(#..): ?

Reply:
Done